src/main.rs

```diff
@@ -0,0 +1,162 @@
+#[macro_use]
+extern crate quicli;
+extern crate dialoguer;
+extern crate git2;
+extern crate ident_case;
+extern crate indicatif;
+extern crate liquid;
+extern crate regex;
+extern crate walkdir;
+
+use dialoguer::Input;
+use git2::{Config as GitConfig, Repository as GitRepository, RepositoryInitOptions};
+use quicli::prelude::*;
+use std::{env, fs};
+use walkdir::WalkDir;
+
+#[derive(Debug, StructOpt)]
+struct Cli {
+    #[structopt(long = "git")]
+    git: String,
+    #[structopt(long = "name")]
+    name: Option<String>,
+}
+
+main!(|args: Cli| {
+    let name = match &args.name {
+        Some(ref n) => n.to_string(),
+        None => query_name()?,
+    };
+
+    let project_dir = env::current_dir()
+        .unwrap_or_else(|_e| ".".into())
+        .join(&name);
+
+    ensure!(
+        !project_dir.exists(),
+        "Target directory `{}` already exists, aborting.",
+        project_dir.display()
+    );
+
+    let _template = GitRepository::clone(&args.git, &project_dir)
+        .with_context(|_e| format!("Couldn't clone `{}`", &args.git))?;
+
+    fs::remove_dir_all(&project_dir.join(".git"))
+        .context("Error cleaning up cloned template")?;
+
+    let engine = liquid::ParserBuilder::new().build();
+    let mut placeholders = liquid::Object::new();
+    placeholders.insert("project-name".to_owned(), liquid::Value::scalar(&name));
```

Comment:
would really prefer these `to_owned`s be `String::from`s or `to_string`s for readability and contributors sake

Reply:
unless this is somehow required for liquid?? 🤔

Reply:
Changed to `String::from`.

I'm super lax with these type conversions, sorry. But you are right and I agree, for other people reading the code, `.to_string()` gives you way more information. This is useful when learning -- you can see that what you are inserting in this line is a `String` and a `liquid::Value`.

But: I don't think you really need to know that. I trust the compiler will tell me if the types don't work, and otherwise I don't really care about if it's a `String` or a `&str` or a `Path(Buf)` or an `OsStr(ing)` or a `Cow<str>` or a `Rc<str>`or whatever. Because… Why would I? This doesn't change the content and it's an API I don't control! Out of curiosity, I hovered over the `placeholders` binding and RLS told me (and [the docs](https://docs.rs/liquid/0.14.3/liquid/type.Object.html) agree): This is a `HashMap<String, Value>` behind a type alias. I had no idea -- and I didn't need. Why am I trying to explain this? I'm not entirely sure. It's something I've been thinking of a lot, how explicit code needs to be.

(In this specific case, I'm also pretty sure I copy-pasted this from the liquid docs. I would've written `into()` because it's faster to type.)